myframe.cpp

```diff
@@ -1077,6 +1089,7 @@ void MyFrame::UpdateButtonsStatus()
     if (!guiding_active ^ m_autoSelectStarMenuItem->IsEnabled())
     {
         m_autoSelectStarMenuItem->Enable(!guiding_active);
+        cond_update_tool(MainToolbar, BUTTON_AUTOSTAR, m_autoSelectStarMenuItem, !guiding_active);

```

Comment:
so as to only trigger the ui refresh when the button state changes should be:
```
if (cond_update_tool(...))
    need_update = true;
```

Reply:
please disregard, I see you already have the chage detection in the outer conditional